// @swc/cli/lib/swc/dirWorker.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = handleCompile;
const slash_1 = __importDefault(require("slash"));
const path_1 = require("path");
const util_1 = require("./util");
const compile_1 = require("./compile");
async function handleCompile(opts) {
    var _a;
    // Create a deep clone of the options to prevent shared references
    const clonedOpts = (0, util_1.deepClone)(opts);
    const dest = (0, util_1.getDest)(clonedOpts.filename, clonedOpts.outDir, clonedOpts.cliOptions.stripLeadingPaths, `.${(_a = clonedOpts.outFileExtension) !== null && _a !== void 0 ? _a : (0, util_1.mapTsExt)(clonedOpts.filename)}`);
    const sourceFileName = (0, slash_1.default)((0, path_1.relative)((0, path_1.dirname)(dest), clonedOpts.filename));
    // Create a fresh copy of the swcOptions
    const options = (0, util_1.deepClone)(clonedOpts.swcOptions);
    // Set sourceFileName in the options
    options.sourceFileName = sourceFileName;
    // Ensure we have the right extension for output files
    // Instead of directly setting on module.outFileExtension (which might not exist in the type),
    // we'll pass it separately to the compile function
    const result = await (0, util_1.compile)(clonedOpts.filename, options, clonedOpts.sync, dest);
    if (result) {
        const destDts = (0, util_1.getDest)(clonedOpts.filename, clonedOpts.outDir, clonedOpts.cliOptions.stripLeadingPaths, `.${(0, util_1.mapDtsExt)(clonedOpts.filename)}`);
        const destSourcemap = dest + ".map";
        await (0, compile_1.outputResult)({
            output: result,
            sourceFile: clonedOpts.filename,
            destFile: dest,
            destDtsFile: destDts,
            destSourcemapFile: destSourcemap,
            options,
        });
        return 1 /* CompileStatus.Compiled */;
    }
    else {
        return 2 /* CompileStatus.Omitted */;
    }
}
